nautobot_ssot/migrations/0007_remove_dashed_custom_fields.py

```diff
@@ -0,0 +1,50 @@
+from django.db import migrations
+from nautobot.extras.utils import FeatureQuery
+
+CF_KEY_CHANGE_MAP = {
+    "servicenow_last_synchronized": "ssot-synced-to-servicenow",
+    "ssot_last_synchronized": "ssot-synced-from-ipfabric",
+    "ipfabric_site_id": "ipfabric-site-id",
+}
+
+
+def migrate_dashed_custom_fields_to_underscored(apps, schema_editor):
+    CustomField = apps.get_model("extras", "customfield")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    for key in CF_KEY_CHANGE_MAP.values():
+        for custom_field in CustomField.objects.filter(key=key):
+            original_key = custom_field.key
+            updated_key = CF_KEY_CHANGE_MAP[key]
+            print(
+                f'CustomField instance "{custom_field.label}" key attribute "{original_key}" is being changed to "{updated_key}".'
+            )
+            custom_field.key = updated_key
+            custom_field.save()
+
+    for ct in ContentType.objects.filter(FeatureQuery("custom_fields").get_query()):
+        relevant_custom_fields = CustomField.objects.filter(content_types=ct)
+        if not relevant_custom_fields.exists():
+            continue
+        model = apps.get_model(ct.app_label, ct.model)
+        cf_list = []
+        for instance in model.objects.all():
+            new_custom_field_data = {}
+            for cf in relevant_custom_fields:
+                new_custom_field_data[cf.key] = instance._custom_field_data.pop(CF_KEY_CHANGE_MAP.get(cf.key), None)
+            instance._custom_field_data = new_custom_field_data
```

Comment:
This probably needs to be reworked since I think as written it's going to throw away any custom field value that *isn't* in CF_KEY_CHANGE_MAP?

Reply:
AH, this was based off the other referenced migration.

Reply:
Yeah, the other migration was explicitly processing *all* CFs and storing them into `CF_KEY_TO_NAME`. Here we're just operating on a subset.

Reply:
Got it.